ml-package-versions.yml

```diff
@@ -0,0 +1,100 @@
+sklearn:
+  package_info:
+    pip_release: "scikit-learn"
+    pip_dev: |
+      pip install git+https://github.com/scikit-learn/scikit-learn.git@master
+
+  models:
+    minimum: "0.23.2"
+    maximum: "0.23.2"
+    run: |
+      pytest tests/sklearn/test_sklearn_model_export.py --large
+
+  autologging:
+    minimum: "0.23.2"
+    maximum: "0.23.2"
+    requirements: ["matplotlib"]
+    run: |
+      pytest tests/sklearn/test_sklearn_autolog.py --large
+
+pytorch:
+  package_info:
+    pip_release: "torch"
+    pip_dev: |
+      conda install pytorch -c pytorch-nightly
+
+  models:
+    minimum: "1.7.0"
+    maximum: "1.7.0"
+    requirements: ["scikit-learn", "torchvision"]
+    run: |
+      pytest tests/pytorch/test_pytorch_model_export.py --large
+
+  autologging:
+    minimum: "1.7.0"
+    maximum: "1.7.0"
+    requirements: ["scikit-learn", "torchvision", "pytorch-lightning"]
+    run: |
+      pytest tests/pytorch/test_pytorch_model_export.py --large
+
+tensorlflow:
+  package_info:
+    pip_release: "tensorflow"
+    pip_dev: |
+      pip install tf-nightly
+
+  models:
+    minimum: "2.3.1"
+    maximum: "2.3.1"
+    run: |
+      pytest tests/tensorflow/test_tensorflow2_model_export.py --large
+
+  autologging:
+    minimum: "2.3.1"
+    maximum: "2.3.1"
+    run: |
+      pytest --verbose tests/tensorflow_autolog/test_tensorflow2_autolog.py --large
+
+xgboost:
+  package_info:
+    pip_release: "xgboost"
+    pip_dev: |
+      git clone --recursive https://github.com/dmlc/xgboost
+      cd xgboost/python-package
+      python setup.py install
+
+  models:
+    minimum: "1.2.1"
+    maximum: "1.2.1"
+    requirements: ["scikit-learn"]
+    run: |
+      pytest tests/xgboost/test_xgboost_model_export.py --large
+
+  autologging:
+    minimum: "1.2.1"
+    maximum: "1.2.1"
+    requirements: ["scikit-learn", "matplotlib"]
+    run: |
+      pytest tests/xgboost/test_xgboost_autolog.py --large
+
+lightgbm:
+  package_info:
+    pip_release: "lightgbm"
+    pip_dev: |
```

Comment:
Should we rename this filed (e.g. `install_dev`)?

Reply:
If we require more complex instructions than `pip install {package_name}` where `{package_name}` is a PyPI-hosted package name or a GitHub URL, then I'd propose we do `install_dev`. We could then change `pip_release` to `package_name`.